### PR TITLE
Add ability to specify custom image size when downloading Site Image

### PR DESCRIPTION
**Pull request by salimbraksa**
Part of #19510

## Related PRs
The following PR should be merged before this one:
- https://github.com/wordpress-mobile/WordPress-iOS/pull/19578

## Context
The following method `UIImageView.downloadSiteIcon(for:placeholderImage:)` is used to download the Site Image. The underlying URL this method is calling also allow the image to be downloaded in a custom size, but this method is hard-coding the size URL param in its implementation details.

The Site Image size in [Migration Welcome Screen](https://github.com/wordpress-mobile/WordPress-iOS/issues/19510) is 60x60, whereas it is 40x40 in other screens. So the purpose of this PR is to allow passing a custom size when calling the `UIImageView.downloadSiteIcon` method.

## Changes
1. I changed `SiteIconDefaults.imageSize` type to `CGSize` and the value to `CGSize(width: 40, height: 40)
2. I added a `imageSize: CGSize` param to `UIImageView.downloadSiteIcon(for:placeholderImage:)` method. Default value is `SiteIconDefaults.imageSize`
3. Updated `UIImageView.downloadSiteIcon` implementation details to read the image size from the `imageSize` param instead of `SiteIconDefaults.imageSize` constant.

## Screenshots

| 60x60 Image Size | 40x40 Image Size |
| ------------------ | ------------------ |
|   |  |

### 1️⃣ Test a screen using the default `imageSize` value.
1. Clean install the Jetpack app and log into your account.
2. Under "My Site" tab, <a href="https://user-images.githubusercontent.com/9609223/200563991-c068e191-448c-4310-b88a-a164ad2a8cf8.png">tap the Sites switcher component</a>
4. You should see a list of Sites.
5. **Expect** the Site Image to be a square of size 40x40. It should look like the 2nd screenshot.

### 2️⃣ Test a screen using a custom `imageSize` value.

<details><summary><b>Prerequisites:</b>  Follow these steps to see the Welcome Screen when the Jetpack app launches</summary>

1. Locate the `shouldShowMigrationUI` computed property in `JetpackWindowManager` class.
2. Change its value to `true`.
</details>

1. Build and install Jetpack app.
2. **If not logged in:** Login then close & re-open the app
3. **Expect** to see the Welcome Screen.
5. **Verify** the Site Image to look slightly larger compared to the previous screen. See the 1st screenshot.

## Regression Notes
1. Potential unintended areas of impact
The `downloadSiteIcon(for:placeholderImage:)` implementation details was slightly modified, so we should test that it is behaving as expected.

4. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual Testing.

6. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Extensions/UIImageView+SiteIcon.swift [hunk 1]
```diff
@@ -61,13 +61,10 @@ extension UIImageView {
...
     ///     - request: the request for the SiteIcon.
-    ///     - placeholderImage: Yes. It's the "place holder image".
+    ///     - config: Includes parameters to take into consideration when downloading the blog image.
     ///
-    private func downloadSiteIcon(
-        with request: URLRequest,
-        placeholderImage: UIImage?) {
-
-        af_setImage(withURLRequest: request, placeholderImage: placeholderImage, completion: { [weak self] dataResponse in
+    private func downloadSiteIcon(with request: URLRequest, config: SiteIconDownloadConfig) {
```
Comment (Gio2018) [inline WordPress/Classes/Extensions/UIImageView+SiteIcon.swift]: I am not opposed to add another method, but how about we use just one and add an `imageSize` parameter which defaults to the default value of 40? This way we don't need another dedicated type and it feels a bit lighter, wdyt?

**Review by Gio2018 — COMMENTED** [review]
Works as expected! Left one code comment for you to consider. Also, if you merge back trunk, the pre-requisite instructions would simplify to just set `shouldShowMigrationUI` to `true`

**Comment by mokagio** [comment]
Given this PR hasn't been approved and also has conflicts with `trunk` I'll push it from 21.2 to 21.3 as I'm getting ready for the 21.2 code freeze on Monday.

**Review by Gio2018 — APPROVED** [review]
LGTM @salimbraksa !

I made a before/after comparison of the site icon images and I can see the difference (see below)

before | after
-- | --
before | after

**Pull request merged**